src/interface/src/app/map/map.component.ts

```diff
@@ -57,51 +46,72 @@ import { PlanCreateDialogComponent } from './plan-create-dialog/plan-create-dial
 import { SignInDialogComponent } from './sign-in-dialog/sign-in-dialog.component';
 import { FeatureService } from '../features/feature.service';
-import { AreaCreationAction, LEGEND } from './map.constants';
-
+import {
+  AreaCreationAction,
+  ERROR_SNACK_CONFIG,
+  LEGEND,
+} from './map.constants';
+import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+@UntilDestroy()
 @Component({
   selector: 'app-map',
   templateUrl: './map.component.html',
   styleUrls: ['./map.component.scss'],
 })
 export class MapComponent implements AfterViewInit, OnDestroy, OnInit, DoCheck {
-  mapManager: MapManager;
+  @Input() planId: string | null = null;
 
-  maps: Map[];
-  mapViewOptions$ = new BehaviorSubject<MapViewOptions>(
-    defaultMapViewOptions()
+  readonly AreaCreationAction = AreaCreationAction;
+  readonly legend: Legend = LEGEND;
+  readonly login_enabled = this.featureService.isFeatureEnabled('login');
+  readonly maps: Map[] = ['map1', 'map2', 'map3', 'map4'].map(
+    (id: string, index: number) => {
+      return {
+        id: id,
+        name: `${index + 1}`,
+        config: defaultMapConfig(),
+      };
+    }
   );
-  mapNameplateWidths: BehaviorSubject<number | null>[] = Array(4)
-    .fill(null)
-    .map((_) => new BehaviorSubject<number | null>(null));
 
-  boundaryConfig$: Observable<BoundaryConfig[] | null>;
-  conditionsConfig$: Observable<ConditionsConfig | null>;
+  mapManager: MapManager;
   regionRecord: string = '';
-  selectedRegion$: Observable<Region | null>;
-  planState$: Observable<PlanState>;
-  selectedMap$: Observable<Map | undefined>;
-
-  existingProjectsGeoJson$ = new BehaviorSubject<GeoJSON.GeoJSON | null>(null);
-
   loadingIndicators: { [layerName: string]: boolean } = {
     existing_projects: true,
   };
-
-  legend: Legend = LEGEND;
-
-  /** Actions bar variables */
-  readonly AreaCreationAction = AreaCreationAction;
+  selectedAreaCreationAction = AreaCreationAction.NONE;
   showUploader = false;
-  selectedAreaCreationAction: AreaCreationAction = AreaCreationAction.NONE;
-  showConfirmAreaButton$ = new BehaviorSubject(false);
-
-  private readonly destroy$ = new Subject<void>();
-  login_enabled = this.featureService.isFeatureEnabled('login');
   drawingLayer: L.GeoJSON | undefined;
 
-  @Input() planId: string | null = null;
+  mapViewOptions$ = new BehaviorSubject<MapViewOptions>(
+    defaultMapViewOptions()
+  );
+  mapNameplateWidths = Array(4)
+    .fill(null)
+    .map((_) => new BehaviorSubject<number | null>(null));
 
+  boundaryConfig$ = this.mapService.boundaryConfig$
+    .asObservable()
+    .pipe(untilDestroyed(this));
+  conditionsConfig$ = this.mapService.conditionsConfig$.asObservable();
+  selectedRegion$ = this.sessionService.region$.asObservable();
+
+  selectedMap$ = this.mapViewOptions$.pipe(
+    map((options) => this.maps[options.selectedMapIndex])
+  );
+  getOpacityForSelectedMap$ = this.selectedMap$.pipe(
```

Comment:
nit:  If this is now a property, should we rename this to `opacityForSelectedMap` or even 'selectedMapOpacity`?

Reply:
yes! good catch